src/styles/getTooltipStyles.js

```diff
@@ -0,0 +1,198 @@
+    componentHeight,
+    tooltipWidth,
+    tooltipHeight,
+    manualShiftHorizontal,
+    manualShiftVertical,
+) {
+    // Determine if the tooltip should display below the wrapped component.
+    // If a tooltip will try to render within GUTTER_WIDTH logical pixels of the top of the screen,
+    // we'll display it beneath its wrapped component rather than above it as usual.
+    const shouldShowBelow = (yOffset - tooltipHeight) < GUTTER_WIDTH;
+
+    // Determine if we need to shift the tooltip horizontally to prevent it
+    // from displaying too near to the edge of the screen.
+    const horizontalShift = computeHorizontalShift(windowWidth, xOffset, componentWidth, tooltipWidth);
+
+    const tooltipVerticalPadding = spacing.pv1;
+    const tooltipFontSize = variables.fontSizeSmall;
+
+    return {
+        animationStyle: {
+            transform: [{
+                scale: currentSize,
+            }],
+        },
+        tooltipWrapperStyle: {
+            position: 'absolute',
+            backgroundColor: themeColors.heading,
+            borderRadius: variables.componentBorderRadiusSmall,
+            ...tooltipVerticalPadding,
+            ...spacing.ph2,
+
+            // Because it uses absolute positioning, the top-left corner of the tooltip is aligned
+            // with the top-left corner of the wrapped component by default.
+            // So we need to shift the tooltip vertically and horizontally to position it correctly.
+            //
+            // First, we'll position it vertically.
+            // To shift the tooltip down, we'll give `top` a positive value.
+            // To shift the tooltip up, we'll give `top` a negative value.
+            top: shouldShowBelow
+
+                // We need to shift the tooltip down below the component. So shift the tooltip down (+) by...
+                ? componentHeight + POINTER_HEIGHT + manualShiftVertical
+
+                // We need to shift the tooltip up above the component. So shift the tooltip up (-) by...
+                : -(tooltipHeight + POINTER_HEIGHT) + manualShiftVertical,
+
+            // Next, we'll position it horizontally.
+            // To shift the tooltip right, we'll give `left` a positive value.
+            // To shift the tooltip left, we'll give `left` a negative value.
+            //
+            // So we'll:
+            //   1) Shift the tooltip right (+) to the center of the component,
+            //      so the left edge lines up with the component center.
+            //   2) Shift it left (-) to by half the tooltip's width,
+            //      so the tooltip's center lines up with the center of the wrapped component.
+            //   3) Add the horizontal shift (left or right) computed above to keep it out of the gutters.
+            //   4) Lastly, add the manual horizontal shift passed in as a parameter.
+            left: ((componentWidth / 2) - (tooltipWidth / 2)) + horizontalShift + manualShiftHorizontal,
```

Comment:
Same question about required parameter `manualShiftHorizontal`. Are we going to always need to pass these props?